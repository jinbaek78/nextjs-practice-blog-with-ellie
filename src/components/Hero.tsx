import Image from 'next/image';
import ProfileImage from '../../public/images/profile.jpeg';
import Link from 'next/link';

export default function Hero() {
  return (
    <section className="text-center">
      <Image
        className="rounded-full mx-auto"
        src={ProfileImage}
        alt="picture of author"
        width={250}
        priority
      />
      <h2 className="text-3xl font-bold mt-2">{"Hi, I'm Jin"}</h2>
      <h3 className="text-xl font-semibold">Full-stack Engineer</h3>
      <Link href={'/contact'}>
        <button className="bg-yellow-500 font-bold rounded-xl py-1 px-4 mt-2">
          Contact Me
        </button>
      </Link>
    </section>
  );
}
