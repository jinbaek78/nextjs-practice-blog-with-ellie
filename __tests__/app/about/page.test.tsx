import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import AboutPage from '@/app/about/page';
import Hero from '@/components/Hero';
import { AiFillDelete } from 'react-icons/ai';

jest.mock('@/components/Hero');

describe.skip('AboutPage', () => {
  afterEach(() => {
    (Hero as jest.Mock).mockReset();
  });
  it('renders', () => {
    const { container } = render(<AboutPage />);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly', () => {
    render(<AboutPage />);

    expect(Hero).toHaveBeenCalled();
  });
});
