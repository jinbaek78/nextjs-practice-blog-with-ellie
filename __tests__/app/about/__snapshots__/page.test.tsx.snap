// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage renders 1`] = `
<div>
  <section
    class="bg-gray-100 shadow-lg p-8 m-8 text-center"
  >
    <h2
      class="text-2xl font-bold text-gray-800 my-2"
    >
      Who AM I?
    </h2>
    <p>
      A FullStack developer who loves development so much 
      <br />
      I am making an web application including human and design
    </p>
    <h2
      class="text-2xl font-bold text-gray-800 my-2"
    >
      Career
    </h2>
    <p>
      Googler (-Now)
      <br />
      maceBook (-2019) 
      <br />
      samjun (-2016)
       
    </p>
    <h2
      class="text-2xl font-bold text-gray-800 my-2"
    >
      Skills
    </h2>
    <p>
      React, Vue, Node
      <br />
      Git, Clean code
      <br />
      VS Code, MongoDB
    </p>
  </section>
</div>
`;
